src/sage/algebras/jordan_algebra.py

```diff
@@ -347,7 +359,10 @@ def gens(self) -> tuple:
             ...
             NotImplementedError: infinite set
         """
-        return tuple(self.algebra_generators())
+        F = self.algebra_generators()
+        if self.is_finite():
```

Comment:
```suggestion
        if F.is_finite():
```
Shouldn’t this be the correct check since we don’t care if the base ring is finite or not (as it is an algebra)?